### PR TITLE
Fixes to 0.5.x

### PR DESCRIPTION
**Pull request by ssunkara1**
1. Fixed tooltip positioning in 0.5.x.
2. Added functions for grids, hline and vline.
3. Fixed bug with clear in pyplot.
4. Modified pyplot example.

Note: This is intended only 0.5.x. Points 3 and 4 are already fixed in 0.6.x. For 1 and 2, a separate PR has to created.

### CODE REVIEW

File: bqplot/nbextension/Mark.js [hunk 1]
```diff
@@ -223,7 +223,8 @@ define(["nbextensions/widgets/widgets/js/widget", "./components/d3/d3",
                 var ref_el = d3.select(document.body).select("#notebook").node();
                 var ref_mouse_pos = d3.mouse(ref_el);
 
-                var mouse_pos = d3.mouse(this.parent.el);
+                var parent_node = this.parent.el.parentElement;
+                var mouse_pos = d3.mouse(parent_node);
```
Comment (dmadeka) [inline bqplot/nbextension/Mark.js]: Does this need to be two lines?

**Comment by dmadeka** [comment]
When I checkout this branch, the plots fail to generate in the notebook.

**Comment by SylvainCorlay** [comment]
Does this apply on master too?

**Comment by dmadeka** [comment]
@ssunkara1 Im good to merge this if you want to add it to the current master as well?

**Comment by ssunkara1** [comment]
I will raise an equivalent PR to master.

**Pull request closed without merging**